MAUI/DataGrid/Grouping.md

```diff
@@ -0,0 +1,572 @@
+            return "Between 10 To 15";
+        if (orderInfo.OrderID >= 15 && orderInfo.OrderID <= 20)
+            return "Between 15 To 20";
+        else
+            return ">20";
+    }
+
+    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+    {
+        return null;
+    }
+}
+
+public class PriceConverter : IValueConverter
+{
+	public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+    {
+        var orderInfo = value as OrderInfo;
+        if (orderInfo.Price < 10)
+            return "<10";
+        if (orderInfo.Price >= 10 && orderInfo.Price < 16)
+            return "Between 10 To 15";
+        if (orderInfo.Price >= 15 && orderInfo.Price <= 20)
+            return "Between 15 To 20";
+        return ">20";
+    }
+
+    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+    {
+        return null;
+    }
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+As you can see in the below screenshot, the records are sorted based on the `OrderNo` column, since `SortGroupRecords` is set as `true` and it is not sorted further based on `Price` column since `SortGroupRecords` is set as `false`.
+
+![SfDataGrid with SortGroupRecords](Images\Grouping\SortMode.png)
+
+## Expand groups while grouping
+ 
+To expand all groups while grouping, set the [SfDataGrid.AutoExpandGroups]() to `true`. While grouping any column, all groups will be in expanded state. 
+
+{% tabs %}
+{% highlight xaml %}
+<syncfusion:SfDataGrid  x:Name="dataGrid"
+                        AutoExpandGroups="True"
+                        AllowGroupExpandCollapse="True"
+                        ItemsSource="{Binding Orders}"/>
+{% endhighlight %}
+{% highlight c# %}
+this.dataGrid.AutoExpandGroups = true;
+this.dataGrid.AllowGroupExpandCollapse = true;
+{% endhighlight %}
+{% endtabs %}
+
+## Expand or collapse the groups
+
+By default, the groups will be in expanded state. However expand or collapse a group in runtime by setting the [SfDataGrid.AllowGroupExpandCollapse]() to `true`.
+
+{% tabs %}
+{% highlight xaml %}
+
+<syncfusion:SfDataGrid  x:Name="dataGrid"
+                        AllowGroupExpandCollapse="True"
+                        ItemsSource="{Binding Orders}"/>
+                       
+{% endhighlight %}
+{% highlight c# %}
+this.dataGrid.AllowGroupExpandCollapse = true;
+{% endhighlight %}
+{% endtabs %}
+
+### Expand or collapse all the groups
+
+Programmatically expand or collapse all the groups at runtime using the [SfDataGrid.ExpandAllGroup]() and [SfDataGrid.CollapseAllGroup]() methods.
+
+{% tabs %}
+{% highlight c# %}
+this.dataGrid.ExpandAllGroup();
+this.dataGrid.CollapseAllGroup();
+{% endhighlight %}
+{% endtabs %}
+
+### Expand or collapse a specific group
+
+Expand or collapse specific group by using the [SfDataGrid.ExpandGroup]() and [SfDataGrid.CollapseGroup]() methods.
```

Comment:
To expand and collapse specfic groups programmatically, you can simply invoke the [SfDataGrid.ExpandAllGroup]() and [SfDataGrid.CollapseAllGroup]() methods.